esp/public/media/scripts/custom_form.js

```diff
@@ -605,37 +669,40 @@ var onSelectElem = function(item) {
 	$j('#id_instructions').val('');
 	$j('#id_required').prop('checked','');
 	
-	$j('div.field_selected').removeClass('field_selected');
-	var currCategory=$j('#cat_selector').val();	
-	var $option,$wrap_option,i, question_text=formElements[currCategory][item]['ques'], $button=$j('#button_add');
-	
-    //  Add validation options
-    if (item in formElements['Generic'])
-    {
-        addCorrectnessOptions(item);
+	if($j('div.field_selected').length){
+        onSelectField($j('div.field_selected'), $j.data($j('div.field_selected')[0], 'data'), item);
+    } else {
+        var currCategory=$j('#cat_selector').val();
+        var $option,$wrap_option,i, question_text=formElements[currCategory][item]['ques'], $button=$j('#button_add');
+        
+        //  Add validation options
+        if (item in formElements['Generic'])
+        {
+            addCorrectnessOptions(item);
+        }
+        
+        //Defining actions for generic elements
+        if(item=='textField' || item=='longTextField' || item=='longAns' || item=='reallyLongAns')
+            addSpecificOptions(item, '', '');
+        else if(item=="radio" || item=="dropdown" || item=="multiselect" || item=="checkboxes") 
+            generateOptions();
+        else if(item=="numeric") 
+            addSpecificOptions(item, '', '');
+        else if(item=='section'){
+            $j('#id_instructions').val('Enter a short description about the section');
+        }
+        else if(item=='page'){
+            $j('#id_instructions').val('Not required');
+        }
+        
+        //Set 'Required' to a sensible default
+        setRequired(item);	
+            
+        $j('#id_question').val(question_text);
+        $prevField=$currSection.children(":last");
+        if($button.val()!='Add to Form')
+            $button.val('Add to Form').html('Add Field to Form').unbind('click').click(function(){insertField($j('#elem_selector').val(),$prevField)});
```

Comment:
When I click the add field to form button, the field options kind of reset, but the field i maybe added doesn't show up in the form preview on the left?